ci/jenkins/utils.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function check_and_upgrade_golang() {
+    echo "====== Checking current Golang version ======"
+    current_version=$(/usr/local/go/bin/go version | grep -oP "go\d+\.\d+(\.\d+)?")
+    project_golang_version=$(go mod edit -json | jq -r .Go)
+    if [ "$current_version" = "go$project_golang_version" ]; then
+        echo "====== Golang version on the testbed is $current_version ======"
+        echo "====== No need to upgrade ======"
+    else
+        echo "====== Upgrading golang version from "$current_version" to "go$project_golang_version.0" ======"
+        install_golang "go${project_golang_version}.0"
```

Comment:
I feel we can't assume that the Go version in Antrea go.mod will always be a format like `1.19`, it's better to check if it already has the patch number or not before add the `.0`.

Reply:
Agreed. I found that the golang releaser didn't keep the version number constantly. Before version 1.21, there's no patch number in the first release, such `go1.20`, not `go1.20.0`. Since 1.21, the first release introduced a patch number, for example `go1.21.0` not `go1.21`.  In order to solve it, everytime we need to download a specific version, we will query `go.dev` and download the latest patch for the minor version.